modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -387,6 +387,62 @@ export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
     return { ...(await keychains.add(recipientKeychainParams)), reducedEncryptedPrv: reducedEncryptedPrv };
   }
 
+  /**
+   * Converts a User or Backup MPCv1 SigningMaterial to RetrofitData needed by MPCv2 DKG.
+   *
+   * @param decryptedKeyshare - MPCv1 decrypted signing material for user or backup as a json.stringify string and bitgo's Big Si.
+   * @param partyId - The party ID of the MPCv1 keyshare.
+   * @returns The retrofit data needed to start an MPCv2 DKG session.
+   */
+  getKeyDataForRetrofit(
+    decryptedKeyshare: string,
+    partyId: MPCv2PartiesEnum.BACKUP | MPCv2PartiesEnum.USER
+  ): DklsTypes.RetrofitData {
+    const mpc = new Ecdsa();
+    const xiList = [
+      Array.from(bigIntToBufferBE(BigInt(1), 32)),
+      Array.from(bigIntToBufferBE(BigInt(2), 32)),
+      Array.from(bigIntToBufferBE(BigInt(3), 32)),
+    ];
+    return this.getMpcV2RetrofitDataFromMpcV1Key({
+      mpcv1PartyKeyShare: decryptedKeyshare,
+      mpcv1PartyIndex: partyId === MPCv2PartiesEnum.USER ? 1 : 2,
+      xiList,
+      mpc,
+    });
+  }
+
+  /**
+   * Initialize a DKG session for MPCv2. If a decrypted keyshare is provided, it will be used to initialize the DKG session for retrofitting MPCv1 keys into MPCv2.
+   * @param m - The number of key shares required to reconstruct the key.
+   * @param n - The total number of key shares.
+   * @param decryptedKeyShare - user or backup decrypted keyshare from MPCv1
+   * @param partyId - The party ID of the MPCv1 keyshare.
+   * @param seed - Random seed for the DKG session.
+   *
+   * @return The initialized DKG session.
+   */
+  initializeDKGSession({
+    m,
+    n,
+    decryptedKeyShare,
+    partyId,
+    seed,
+  }: {
+    m: number;
+    n: number;
+    decryptedKeyShare?: string;
+    partyId: MPCv2PartiesEnum.USER | MPCv2PartiesEnum.BACKUP;
+    seed?: Buffer;
+  }): DklsDkg.Dkg {
+    if (decryptedKeyShare) {
+      const retrofitData = this.getKeyDataForRetrofit(decryptedKeyShare, partyId);
+      return new DklsDkg.Dkg(n, m, partyId, seed, retrofitData);
```

Comment:
i guess you would also want to dependency inject a dkls wasm since im assuming this helper is to be called from applications that will require the bundler wasm rather than the default node wasm

Reply:
hmm good point, in that case I think it would be better to just have the caller for DKG to init the session themselves and manage how the wasm should be loaded. In which case this wouldn't be needed.